sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessCreate.java

```diff
@@ -204,6 +209,32 @@ public T getCurrent() throws NoSuchElementException {
 
       @Override
       public void close() throws IOException {}
+
+      @Override
+      public Double getFractionConsumed() {
+        return rangeTracker.getFractionConsumed();
+      }
+
+      @Override
+      public InMemorySource<T> splitAtFraction(double fraction) {
+        long splitPosition = rangeTracker.getPositionForFractionConsumed(fraction);
+        if (rangeTracker.trySplitAtPosition(splitPosition)) {
```

Comment:
invert the `if` to simplify code -- if it fails, return null. then un-indent and actuate the split.